esy-build

```diff
@@ -1,12 +1,11 @@
-#! /bin/bash
+#! /bin/sh
 
 # esy-build
 #
 # Wrapper to execute appropriate build strategy, based on platform
 
 set -u
 set -e
-set -o pipefail
```

Comment:
It is not a big loss since we don't use pipe. Don't know if `set -o pipe fail || true` work on POSIX sh. Might be desirable if in future we use pipe. If it fail shellcheck give workaround. So really not an issue.

Reply:
@EduardoRFS What do you think of this?

Thanks for running these script through shellcheck btw, @Et7f3 !

Reply:
Yeah I would also remove the pipefail property here, I even believe that `set -e` actually makes it redundant.

Reply:
No they serve different goal. When you write a pipe a new instance of bash is launched call pipe/dup... and exec each subcommand in a new process then it wait all and return last exit code. with pipefail option it give a error code if any command in the pipeline fail. See this demo:

```sh
#!/bin/sh

if false | true
then
  echo $?
else
  echo $?
fi
# $? = 0 = return code of true
set -o pipefail
if false | true
then
  echo $?
else
  echo $?
fi
# $? = 1 = because false failed
set +o pipefail
set -e
if false | true
then
  echo $?
else
  echo $?
fi
# $? = 0 because false | true is a new shell that returned exit code of true the last command
```

It produce: 
```sh
$ sh shell.sh
0
1
0
```

The if are here to avoid sh exiting and loose $? value

Reply:
Oh that's nice, didn't knew it, but, anyway this command doesn't go over any pipe.